Model/Plugins/MethodList.php

```diff
@@ -0,0 +1,210 @@
+        \Payone\Core\Helper\Consumerscore $consumerscoreHelper,
+        \Magento\Checkout\Model\Session $checkoutSession
+    ) {
+        $this->consumerscore = $consumerscore;
+        $this->consumerscoreHelper = $consumerscoreHelper;
+        $this->checkoutSession = $checkoutSession;
+    }
+
+    /**
+     * Filter methods by the worst score
+     *
+     * @param  MethodInterface[] $aPaymentMethods
+     * @param  string            $sWorstScore
+     * @return MethodInterface[]
+     */
+    protected function filterMethodsByScore($aPaymentMethods, $sWorstScore)
+    {
+        if ($sWorstScore == 'G') {
+            return $aPaymentMethods;
+        }
+
+        $aYellowMethods = $this->consumerscoreHelper->getAllowedMethodsForScore('Y');
+        $aRedMethods = $this->consumerscoreHelper->getAllowedMethodsForScore('R');
+
+        $aReturnMethods = [];
+        foreach ($aPaymentMethods as $oMethod) {
+            if ($sWorstScore == 'Y' &&
+                (array_search($oMethod->getCode(), $aYellowMethods) !== false ||
+                array_search($oMethod->getCode(), $aRedMethods) !== false)) {
+                $aReturnMethods[] = $oMethod;
+            }
+
+            if ($sWorstScore == 'R' && array_search($oMethod->getCode(), $aRedMethods) !== false) {
+                $aReturnMethods[] = $oMethod;
+            }
+        }
+        return $aReturnMethods;
+    }
+
+    /**
+     * Execute a consumerscore request to PAYONE or load an old score if its lifetime is still active
+     *
+     * @param  AddressInterface $oShipping
+     * @return string
+     */
+    protected function getScoreByCreditrating(AddressInterface $oShipping)
+    {
+        $aResponse = $this->consumerscore->sendRequest($oShipping);
+        if ($aResponse === true) {// creditrating not executed because of a previous check
+            $this->consumerscoreHelper->copyOldStatusToNewAddress($oShipping);
+        }
+
+        if (isset($aResponse['score'])) {
+            $oShipping->setPayoneProtectScore($aResponse['score'])->save();
+        }
+
+        $sScore = $oShipping->getPayoneProtectScore();
+        return $sScore;
+    }
+
+    /**
+     * Get parameter from config
+     *
+     * @param  string $sParam
+     * @param  bool   $blIsAddresscheck
+     * @return string
+     */
+    protected function getConfigParam($sParam, $blIsAddresscheck = false)
+    {
+        $sGroup = 'creditrating';
+        if ($blIsAddresscheck === true) {
+            $sGroup = 'address_check';
+        }
+        return $this->consumerscoreHelper->getConfigParam($sParam, $sGroup, 'payone_protect');
+    }
+
+    /**
+     * Get quote object from session
+     *
+     * @return \Magento\Quote\Model\Quote
+     */
+    protected function getQuote()
+    {
+        return $this->checkoutSession->getQuote();
+    }
+
+    /**
+     * Determine if creditrating is needed
+     *
+     * @return bool
+     */
+    protected function isCreditratingNeeded()
```

Comment:
This Method is too complex, please refactor: https://scrutinizer-ci.com/g/PAYONE-GmbH/magento-2/inspections/f57e816f-b8e5-4e29-b0a2-aa3d5399368d/code-structure/operation/Payone%5CCore%5CModel%5CPlugins%5CMethodList%3A%3AisCreditratingNeeded